cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs

```diff
@@ -0,0 +1,298 @@
+
+  let queryAnyUtxo :: Text.Text -> Integration TxIn
+      queryAnyUtxo address = withFrozenCallStack $ do
+        utxos <- queryUtxos execConfig work utxoFileCounter sbe address
+        H.noteShow =<< H.headM (Map.keys utxos)
+      socketName' = IO.sprocketName poolSprocket1
+      socketBase = IO.sprocketBase poolSprocket1 -- /tmp
+      socketPath = socketBase </> socketName'
+
+  H.note_ $ "Sprocket: " <> show poolSprocket1
+  H.note_ $ "Abs path: " <> tempAbsBasePath'
+  H.note_ $ "Socketpath: " <> socketPath
+
+  -- Create Conway constitution
+  gov <- H.createDirectoryIfMissing $ work </> "governance"
+
+  let stakeVkeyFp = gov </> "stake.vkey"
+      stakeSKeyFp = gov </> "stake.skey"
+
+  _ <- P.cliStakeAddressKeyGen tempAbsPath'
+         $ P.KeyNames { P.verificationKeyFile = stakeVkeyFp
+                      , P.signingKeyFile = stakeSKeyFp
+                      }
+
+  let drepVkeyFp :: Int -> FilePath
+      drepVkeyFp n = tempAbsPath' </> "drep-keys" </> ("drep" <> show n) </> "drep.vkey"
+
+      drepSKeyFp :: Int -> FilePath
+      drepSKeyFp n = tempAbsPath' </> "drep-keys" </> ("drep" <> show n) </> "drep.skey"
+
+  -- Create Drep registration certificates
+  let drepCertFile :: Int -> FilePath
+      drepCertFile n = gov </> "drep-keys" <>"drep" <> show n <> ".regcert"
+  H.forConcurrently_ [1..3] $ \n -> do
+    H.noteM_ $ H.execCli' execConfig
+       [ "conway", "governance", "drep", "registration-certificate"
+       , "--drep-verification-key-file", drepVkeyFp n
+       , "--key-reg-deposit-amt", show @Int 0
+       , "--out-file", drepCertFile n
+       ]
+
+  txin1 <- queryAnyUtxo . paymentKeyInfoAddr $ head wallets
+
+  -- Submit registration certificates
+  drepRegTxbodyFp <- H.note $ work </> "drep.registration.txbody"
+  drepRegTxSignedFp <- H.note $ work </> "drep.registration.tx"
+
+  H.noteM_ $ H.execCli' execConfig
+    [ "conway", "transaction", "build"
+    , "--tx-in", Text.unpack $ renderTxIn txin1
+    , "--change-address", Text.unpack $ paymentKeyInfoAddr $ head wallets
+    , "--certificate-file", drepCertFile 1
+    , "--certificate-file", drepCertFile 2
+    , "--certificate-file", drepCertFile 3
+    , "--witness-override", show @Int 4
+    , "--out-file", drepRegTxbodyFp
+    ]
+
+  H.noteM_ $ H.execCli' execConfig
+    [ "conway", "transaction", "sign"
+    , "--tx-body-file", drepRegTxbodyFp
+    , "--signing-key-file", paymentSKey $ paymentKeyInfoPair $ head wallets
+    , "--signing-key-file", drepSKeyFp 1
+    , "--signing-key-file", drepSKeyFp 2
+    , "--signing-key-file", drepSKeyFp 3
+    , "--out-file", drepRegTxSignedFp
+    ]
+
+  H.noteM_ $ H.execCli' execConfig
+    [ "conway", "transaction", "submit"
+    , "--tx-file", drepRegTxSignedFp
+    ]
+
+  let sizeBefore = 3
+
+  waitDRepsNumber (Api.File configurationFile) (Api.File socketPath) execConfig sizeBefore
+
+  -- While we're at it, let's test conway query drep-state, because we don't do it elsewhere
```

Comment:
So this is good but it should be a separate property where we potentially execute all of our cli queries in a single test.

Reply:
Removed and created follow-up issue: https://github.com/IntersectMBO/cardano-node/issues/5672